### PR TITLE
Chore: Type improvements

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- type improvements in some datasource code

**Why do we need this feature?**

- 🧹 

**Who is this feature for?**

- grafana devs

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/plugins/datasource/loki/datasource.ts [hunk 1]
```diff
@@ -759,7 +761,7 @@ export class LokiDatasource
    * Handles escaping of special characters based on variable type and value.
    * @returns The interpolated value with appropriate character escaping.
    */
-  interpolateQueryExpr(value: any, variable: any) {
+  interpolateQueryExpr(value: any, variable: QueryVariableModel | CustomVariableModel) {
```
Comment (matyax) [inline public/app/plugins/datasource/loki/datasource.ts:764]: <3

**Review by matyax — COMMENTED** [review]
👍  Thank you!

**Review by fabrizio-grafana — APPROVED** [review]
Approved for Tempo plugin.

**Review by baldm0mma — APPROVED** [review]
Looks good from a `Transformers/data-viz` perspective. 👍

**Pull request merged**